### ISSUE TITLE
Firebase Database iOS plugin, if query using startAt or endAt does not specify key value, it causes an exception

### ISSUE THREAD

**Issue by KevinTheGray** [flutter/flutter#10899]
## Steps to Reproduce
- Use version 0.0.9
- if you use a query with startAt/endAt and don't specify a key and use the query.

My code was something like this, this is **NOT** the only case it will happen, but just an example of when it would happen.  I assume it will happen any time you use the query to listen for an event.
```dart
Query query = itemsReference
              .orderByChild("published-date")
              .startAt(
                mostRecentItem.publishedDate + 1,
              )
              .endAt(
                nowInUTCMillis,
              );
query.onValue.listen((Event event) {...
// crashes here
```
## Logs

```
-[NSNull length]: unrecognized selector sent to instance 0x1b44ccef8
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSNull length]: unrecognized selector sent to instance 0x1b44ccef8'
```

## Flutter Doctor
```
[✓] Flutter (on Mac OS X 10.12.4 16E195, locale en-US, channel master)
    • Flutter at /Users/KG/Developer/Flutter/flutter
    • Framework revision 2052741ab3 (29 hours ago), 2017-06-20 11:32:40 -0700
    • Engine revision fffe502d43
    • Tools Dart version 1.24.0-dev.6.7

[✓] Android toolchain - develop for Android devices (Android SDK 26.0.0)
    • Android SDK at /Users/KG/Library/Android/sdk
    • Platform android-26, build-tools 26.0.0
    • ANDROID_HOME = /Users/KG/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 8.3.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 8.3.3, Build version 8E3004b
    • ios-deploy 1.9.1
    • CocoaPods version 1.2.1

[✓] Android Studio (version 2.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Gradle version 3.2
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] IntelliJ IDEA Community Edition (version 2017.1.4)
    • Flutter plugin version 14.0
    • Dart plugin version 171.4694.29
```

**Comment by eseidelGoogle** [comment]
@collinjackson

**Comment by KevinTheGray** [comment]
@collinjackson Sorry if I'm missing something here, but how would this PR fix this?  None of the files modified were dart or objective-c, and this error is thrown from the native iOS side with `-[NSNull length]: unrecognized selector sent to instance 0x1b44ccef8`

**Comment by collinjackson** [comment]
Good catch, I forgot to include my changes to the iOS side to the CL. PTAL.

**Comment by collinjackson** [comment]
Fixed in 0.0.10

**Comment by collinjackson** [comment]
@KevinTheGray was still seeing this issue so I'm reopening and investigating

**Comment by collinjackson** [comment]
Fixed (hopefully for reals this time) in 0.0.11

**Comment by kroikie** [comment]
@KevinTheGray

This issue has been moved to https://github.com/FirebaseExtended/flutterfire/issues/990. Any further collaboration will be done there.
